src/mirakuru/base.py

```diff
@@ -43,53 +49,94 @@
+"""_sre.SRE_Pattern matching PIDs in result from `$ ps xe -o pid,cmd`."""
+
+
+def processes_with_env_psutil(env_name, env_value):
+    """
+    Find PIDs of processes having environment variable matching given one.
+
+    Internally it uses `psutil` library.
+
+    :param str env_name: name of environment variable to be found
+    :param str env_value: environment variable value
+    :return: process identifiers (PIDs) of processes that have certain
+             environment variable equal certain value
+    :rtype: set
+    """
+    pids = set()
 
+    for proc in psutil.process_iter():
+        try:
+            pinfo = proc.as_dict(attrs=['pid', 'environ'])
+        except psutil.NoSuchProcess:
+            pass
+        else:
+            penv = pinfo.get('environ')
+            if penv and penv.get(env_name) == env_value:
+                pids.add(pinfo['pid'])
 
-def processes_with_env(env_name, env_value):
+    return pids
+
+
+def processes_with_env_ps(env_name, env_value):
     """
     Find PIDs of processes having environment variable matching given one.
 
-    Function uses `$ ps e -ww` command so it works only on systems
+    It uses `$ ps xe -o pid,cmd` command so it works only on systems
     having such command available (Linux, MacOS). If not available function
     will just log error.
 
     :param str env_name: name of environment variable to be found
     :param str env_value: environment variable value
-    :return: process ids (PIDs) of processes that have certain environment
-             variable with certain value
+    :return: process identifiers (PIDs) of processes that have certain
+             environment variable equal certain value
     :rtype: set
     """
     pids = set()
-
+    ps_xe = ''
     try:
-        ps_xe = subprocess.check_output(('ps', 'xe', '-ww')).splitlines()
+        cmd = 'ps', 'xe', '-o', 'pid,cmd'
+        ps_xe = subprocess.check_output(cmd).splitlines()
     except OSError as err:
         if err.errno == 2:
-            log.error("`$ ps xe -ww` command was called but it is not "
+            log.error("`$ ps xe -o pid,cmd` command was called but it is not "
                       "available on this operating system. Mirakuru will not "
                       "be able to list the process tree and find if there are "
                       "any leftovers of the Executor.")
-    else:
-        env = '{0}={1}'.format(env_name, env_value)
+            return pids
+    except subprocess.CalledProcessError:
+        log.error("`$ ps xe -o pid,cmd` command exited with non-zero code.")
 
-        for line in ps_xe:
-            line = str(line)
-            if env in line:
-                pids.add(int(PS_XE_PID_MATCH.match(line).group(1)))
+    env = '{0}={1}'.format(env_name, env_value)
+
+    for line in ps_xe:
+        line = str(line)
+        if env in line:
+            pids.add(int(PS_XE_PID_MATCH.match(line).group(1)))
     return pids
 
 
+if psutil_importable is True:
```

Comment:
can we actually separate `processes_with_env` this into a separate module? It would make the base much cleaner